modules/internal/ChapelArray.chpl

```diff
@@ -3957,6 +3960,17 @@ module ChapelArray {
       a <=> b;
   }
 
+  // Sometimes we really need the result of '!' to be an array.
```

Comment:
When in particular do we need that? In my view, this function simply serves to add an additional temporary in some cases.